src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala

```diff
@@ -33,27 +34,32 @@ trait OAuthService extends HttpService with PerRequestCreator with FireCloudDire
   val routes: Route =
     path("login") {
       get {
-        parameter("path") { userpath =>
-          // TODO: future story: generate and persist unique token along with the user path
-          val state = userpath
-
-          // approval prompt - "force" to force a refresh token
-          // TODO: future story: make dynamic based on whether or not the user has a refresh token stored already
-          val approvalPrompt = "force"
+        parameters("path".?, "prompt".?, "callback".?) { (userpath, prompt, callback) =>
```

Comment:
Option: the `callback` parameter is a URL-encoded:
```
https://www.example.com/whatever-i-want/somepath#more-stuff?my-access-token-param={access_token}
```
You store this entire thing in `state` below. Then...